imsim/lsst_image.py

```diff
@@ -0,0 +1,169 @@
+            CCD in the focal plane.
+        """
+        bbox = det.getBBox()
+        nx = bbox.getWidth()
+        ny = bbox.getHeight()
+
+        # Compute the location of the lower-left corner pixel of the
+        # CCD in focal plane coordinates.
+        pix_to_fp = det.getTransform(cameraGeom.PIXELS,
+                                     cameraGeom.FOCAL_PLANE)
+        pixel_size = det.getPixelSize()
+        llc = pix_to_fp.applyForward(lsst.geom.Point2D(0, 0))
+
+        # Generate a grid of pixel x, y locations in the focal plane.
+        xarr, yarr = np.meshgrid(np.arange(nx)*pixel_size.x + llc.x,
+                                 np.arange(ny)*pixel_size.y + llc.y)
+
+        # Compute the radial distance of each pixel to the focal plane
+        # center.
+        r = np.sqrt(xarr**2 + yarr**2)
+
+        return self.spline_model(r)/self.value_at_zero
+
+
+class LSST_ImageBuilder(ScatteredImageBuilder):
+
+    # This is mostly the same as the GalSim "Scattered" image type.
+    # So far the only change is in the sky background image.
+
+    def setup(self, config, base, image_num, obj_num, ignore, logger):
+        """Do the initialization and setup for building the image.
+
+        This figures out the size that the image will be, but doesn't actually build it yet.
+
+        Parameters:
+            config:     The configuration dict for the image field.
+            base:       The base configuration dict.
+            image_num:  The current image number.
+            obj_num:    The first object number in the image.
+            ignore:     A list of parameters that are allowed to be in config that we can
+                        ignore here. i.e. it won't be an error if these parameters are present.
+            logger:     If given, a logger object to log progress.
+
+        Returns:
+            xsize, ysize
+        """
+        # This is mostly a copy of the ScatteredImageType.setup function.
+        logger.debug('image %d: Building LSST_Image: image, obj = %d,%d',
+                     image_num,image_num,obj_num)
+
+        self.nobjects = self.getNObj(config, base, image_num, logger=logger)
+        logger.debug('image %d: nobj = %d',image_num,self.nobjects)
+
+        # These are allowed for LSST_Image, but we don't use them here.
+        extra_ignore = [ 'image_pos', 'world_pos', 'stamp_size', 'stamp_xsize', 'stamp_ysize',
+                         'nobjects' ]
+        opt = { 'size': int , 'xsize': int , 'ysize': int, 'dtype': None,
+                'apply_vignetting': bool, 'apply_sky_gradient': bool,
+                'vignetting_data_file': str}
+        params = GetAllParams(config, base, opt=opt, ignore=ignore+extra_ignore)[0]
+
+        size = params.get('size',0)
+        full_xsize = params.get('xsize',size)
+        full_ysize = params.get('ysize',size)
+
+        self.apply_vignetting = params.get('apply_vignetting', False)
+        vignetting_data_file =  params.get('vignetting_data_file', False)
```

Comment:
False is the wrong default here.  None maybe?
Or better probably, only read it if apply_vignetting is True, in which case, make it required with no default.